File: Serwer/Question.cpp
//
// Created by blazej on 27.04.2020.
//

#include "Question.h"

const std::string &Question::getQuestion() const {
    return question;
}

void Question::setQuestion(const std::string &question) {
    Question::question = question;
}

const std::vector<std::string> &Question::getAnswers() const {
    return answers;
}

void Question::setAnswers(const std::vector<std::string> &answers) {
    Question::answers = answers;
}

char Question::getCorrectAnswer() const {
    return correctAnswer;
}

void Question::setCorrectAnswer(char correctAnswer) {
    Question::correctAnswer = correctAnswer;
}

File: Klient/src/main/java/app/Question.java
package app;

import javafx.collections.FXCollections;

import java.util.List;

/**
 * Klasa przechowująca dane nt. aktualnego pytania otrzymywane z serwera.
 * Fields:
 *  question - pytanie w danej rundzie.
 *  answers - Lista przechowująca odpowiedzi do pytania.
 *  yourAnswer - numer odp. na które dany gracz odpowiedział pyatnia
 */
public class Question {
    private  String question;
    private final List<String> answers;

    public String getQuestion() {
        return question;
    }

    public List<String> getAnswers() {
        return answers;
    }

    public int getCorrectAnswer() {
        return correctAnswer;
    }

    public Question(String question, String a, String b, String c, String d, int correctAnswer) {
        this.answers = FXCollections.observableArrayList();
        this.question = question;
        this.answers.addAll(List.of(a,b,c,d));
        this.correctAnswer = correctAnswer;
    }

    private int correctAnswer;

    public Question() {
        answers = FXCollections.observableArrayList();
    }

    @Override
    public String toString() {
        return this.question + " " + correctAnswer;
    }
}
